### PR TITLE
feat: Update cluster onboarding guide to support monitoring and better logical flow

### PR DESCRIPTION
**Pull request by tumido**
Resolves: https://github.com/operate-first/hitchhikers-guide/issues/84
Follow up to #15 

This PR changes the flow of the guide to:

1. Setup all variables
2. Clone apps repo
3. Enable ArgoCD in ACM
4. Enable SSO
5. Create cluster admin group since we have SSO and users
6. Create cluster scope overlay
7. Enable monitoring
8. Create ArgoCD apps for all above
9. Commit

Example PR created by executing the notebook https://github.com/tumido/apps/commit/e4157cfe670122b43e2a7cc0c8fd44eded3a1eca

### CODE REVIEW

**Comment by mimotej** [comment]
Looks good thank you for changes :+1: 
/lgtm

**Review by mimotej — APPROVED** [review]

**Review by 4n4nd — COMMENTED** [review]
Small nits, other than that 
/lgtm

File: pages/onboarding_new_cluster.ipynb [hunk 1]
```diff
@@ -2,238 +2,196 @@
...
-    "## Changing the apps repository\n",
+    "- [ ] Operate First's ArgoCD can manage the cluster\n",
+    "- [ ] Operate First's SSO is used as identity provider by OpenShift\n",
+    "- [ ] Operate First's integrated metrics and alerting federation is deployed to the cluster\n",
     "\n",
-    "In this part of tutorial we prepare all files which are necessary for onboarding cluster in `apps` repository.\n",
+    "## Introduction\n",
     "\n",
-    "### Workspace setup\n",
+    "All manifests for all the workloads owned by Operate First Ops team are maintained in the `operate-first/apps` repository following the ][Kustomize best practices](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/).\n",
```
[the remark this answers is not in the document]
Comment (tumido) [reply pages/onboarding_new_cluster.ipynb]: Resolved

File: pages/onboarding_new_cluster.ipynb [hunk 2]
```diff
@@ -2,238 +2,196 @@
...
+   ]
   },
   {
    "cell_type": "markdown",
+   "metadata": {},
    "source": [
+    "### 4. Enable SSO login\n",
     "\n",
-    "### Enable ArgoCD management in ACM\n",
+    "Next on the list of tasks that need to happen is to enable SSO for this cluster. Operate First SSO provides users an unified and seamless experience when accessing the cluster. To enable it, we need to setup 2 things:\n",
```
[the remark this answers is not in the document]
Comment (tumido) [reply pages/onboarding_new_cluster.ipynb]: Resolved

**Comment by tumido** [comment]
Updated to decouple `DNS_LOCATION` and `CLUSTER_REGION`. This is the result https://github.com/tumido/apps/commit/9252d2d73e98c6c2aaa5102b000b96892484788e

**Review by 4n4nd — COMMENTED** [review]
/lgtm

File: pages/onboarding_new_cluster.ipynb [hunk 3]
```diff
@@ -404,169 +339,209 @@
...
    "cell_type": "markdown",
+   "metadata": {},
    "source": [
-    "Yet again we need to tell Kustomize to import these additional resources to the new cluster. \\\n",
-    "To do so, please list secret-generator, cluster patch and configuration of SSO in `cluster-scope/overlays/prod/$CLUSTER_REGION/$CLUSTER_NAME/kustomization.yaml`:"
-   ],
-   "metadata": {}
+    "### 5. Create a cluster admins group\n",
+    "\n",
+    "Now we can assume we have Operate First SSO enabled on the cluster, hence we can start using GitHub accounts as user names on the cluster. Let's use that to declare cluster admins for this particular cluster. While cluster admins have full access to the cluster, all changes should always be done via GitOps. In general, we thing of cluster admins as an emergency break in case we need to investigate or act quickly.\n",
```
Comment (HumairAK) [inline pages/onboarding_new_cluster.ipynb:352]: >  In general, we thing of cluster admins as

change `thing` to `think`

File: pages/onboarding_new_cluster.ipynb [hunk 4]
```diff
@@ -404,169 +339,209 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 7. Enable monitoring and alerting\n",
     "\n",
-    "Please fork/clone the [operate-first/argocd-apps](https://github.com/operate-first/argocd-apps) repository. During this whole setup, we’ll be working within this repository.\n",
+    "We follow the recommended practices by OpenShift upstream, that means we support User Workload Monitoring on our clusters. And since Operate First is a community cloud, we aiming to be transparent about alerts fired by the cluster itself. In this step we will enable User Workload Monitoring, then we'll bring in an alert receiver for GitHub that funnels alerts from the cluster and files them as GitHub Issues.\n",
```
Comment (HumairAK) [inline pages/onboarding_new_cluster.ipynb:441]: > we aiming 

to

> we are aiming

File: pages/onboarding_new_cluster.ipynb (hunk 3 above)
Comment (tumido) [reply pages/onboarding_new_cluster.ipynb]: updated

File: pages/onboarding_new_cluster.ipynb (hunk 4 above)
Comment (tumido) [reply pages/onboarding_new_cluster.ipynb]: updated

File: pages/onboarding_new_cluster.ipynb [hunk 5]
```diff
@@ -555,18 +507,31 @@
...
     "\n",
-    "Create `app-of-apps/app-of-apps-$CLUSTER_NAME.yaml`"
+    "\n",
+    "#### 1. Create the App-of-apps\n",
+    "\n",
+    "First we will create the app-of-apps application for this cluster. It's an application which points to other application manifests. This pattern allows us to automate deployment of future ArgoCD applications that we will want to deploy to this cluster.\n",
+    "\n",
+    "Since we assume ArgoCD to be yet another regular application we host manifests which helps to configure it in the `argocd` folder. We host a single instance of it, so we'll be working within the `argocd/overlays/moc-infra/`\n",
+    "\n",
+    "In this step we will create an `Application` resource which points to `argocd/overlays/moc-infra/applications/envs/$CLUSTER_REGION/$CLUSTER_NAME`. That is exactly where we will keep all other `Application` resources for this cluster. Once the app-of-apps resource manifest is create, we'll add it to the Kustomization at `argocd/overlays/moc-infra/applications/app-of-apps/kustomization.yaml`."
```
Comment (HumairAK) [inline pages/onboarding_new_cluster.ipynb:528]: > manifest is create 

to 

> manifest is created

File: pages/onboarding_new_cluster.ipynb [hunk 6]
```diff
@@ -606,35 +574,36 @@
...
   {
    "cell_type": "markdown",
    "source": [
+    "#### 2. Application for privileged resource\n",
     "\n",
-    "### Cluster management application\n",
-    "\n",
-    "As the next step we have to create an environment folder for ArgoCD application resource, where we define resources for this specific cluster. In this folder we can create a new application for cluster management. \n",
-    "As you can see in the application manifest below in the `.spec.source`, the application points to the resources we created in the first section of this tutorial. The path is pointing ArgoCD to the `cluster-scope/overlays/prod/$CLUSTER_REGION/$CLUSTER_NAME` which is precisely where our cluster admin groups and OAuth changes are located.\n"
+    "Now let's add our first application to this folder will. This application should source the overlay in the `cluster-scope` folder, which we've created at **6. Stitch things together via Kustomize** step."
```
Comment (HumairAK) [inline pages/onboarding_new_cluster.ipynb:597]: >  to this folder will. 

I think you meant to omit `will` here?

File: pages/onboarding_new_cluster.ipynb (hunk 5 above)
Comment (tumido) [reply pages/onboarding_new_cluster.ipynb:528]: updated

File: pages/onboarding_new_cluster.ipynb (hunk 6 above)
Comment (tumido) [reply pages/onboarding_new_cluster.ipynb:597]: yeah, updated

**Pull request merged**